### PR TITLE
Add support for RTE dataset with unit tests

### PR DESCRIPTION
**Pull request by vcm2114**
## Summary
- Added support for RTE dataset
- Added mocked unit tests for RTE dataset 

## Test
- ` pytest test/datasets/test_rte.py`

## Context
See #1710

### CODE REVIEW

File: torchtext/datasets/rte.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    url_dp = IterableWrapper([URL])
+    cache_compressed_dp = url_dp.on_disk_cache(
+        filepath_fn=_filepath_fn,
+        hash_dict={_filepath_fn(URL): MD5},
+        hash_type="md5",
+    )
+    cache_compressed_dp = HttpReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(filepath_fn=_extracted_filepath_fn)
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b").read_from_zip().filter(_filter_fn)
```
Comment (Nayef211) [inline torchtext/datasets/rte.py:88]: Same comment as https://github.com/pytorch/text/pull/1717/files#r877330153

**Review by Nayef211 — APPROVED** [review]
LGTM

File: torchtext/datasets/rte.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return os.path.join(root, _EXTRACTED_FILES[split])
+
+    def _filter_fn(x):
+        return _EXTRACTED_FILES[split] in x[0]
+
+    def _modify_res(x):
+        if split == "test":
+            return (x[1], x[2])
+        else:
+            return (int(x[3]), x[1], x[2])
```
Comment (Nayef211) [inline torchtext/datasets/rte.py:77]: Same comment as https://github.com/pytorch/text/pull/1724#discussion_r877338959

**Review by parmeet — APPROVED** [review]
LGTM! Let's address the comments before landing :)

**Pull request merged**